Dockerfile.gsutil

```diff
@@ -11,12 +11,21 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-FROM golang:1.16 as app
-COPY . /go/src/github.com/ossf/scorecard
-WORKDIR /go/src/github.com/ossf/scorecard
-RUN [ "go", "build", "."]
+
+# syntax = docker/dockerfile:1-experimental
+
+FROM  golang:1.16 as base
+WORKDIR /src
+ENV CGO_ENABLED=0
+COPY go.* ./
+RUN go mod download
+COPY . ./
+
+FROM base AS build
+ARG TARGETOS
+ARG TARGETARCH
+RUN CGO_ENABLED=0 go build -a -tags netgo -ldflags '-w -extldflags "-static"' -o /out/scorecard .
```

Comment:
isnt there a make target we can use here ?

Reply:
Agree, we can use `make build`, I will fix it. It was copied from the previous docker file.

Reply:
I am not going to use the `make build` target now. That would require lots of changes to test and other places. So that will be another PR. So let's use this for now.

Reply:
SG.